sphinx/themes/basic/layout.html

```diff
@@ -87,7 +87,6 @@ <h3>{{ _('Navigation') }}</h3>
 {%- endmacro %}
 
 {%- macro script() %}
-    <script type="text/javascript" id="documentation_options" data-url_root="{{ pathto('', 1) }}" src="{{ pathto('_static/documentation_options.js', 1) }}"></script>
```

Comment:
As commented above, 3rd generation themes does not use `js_tag()` helper. So this would not be moved.

In addition, to support this tag by `js_tag()`, we have to give a special code for this like you modified. I feel it is too much.